metapool/prep.py

```diff
@@ -842,3 +846,62 @@ def qiita_scrub_name(name):
         the sample name, formatted for qiita
     """
     return re.sub(r'[^0-9a-zA-Z\-\.]+', '.', name)
+
+
+def _demux_by_project(frame):
+    # use PlateReplication object to convert each sample's 384 well location
+    # into a 96-well location + quadrant. Since replication is performed at
+    # the plate-level, this will identify which replicants belong in which
+    # new sample-sheet.
+    plate = PlateReplication(None)
+
+    frame['quad'] = [plate.get_96_well_location_and_quadrant(x)[0] for x in
+                     list(frame.destination_well_id)]
+
+    gb = frame.groupby('quad')
+
+    return [gb.get_group(grp).drop(['quad'], axis=1) for grp in gb.groups]
```

Comment:
I think these 2 lines are still doing an operation that can happen in a single line:
```
return [group_df.drop(['quad'], axis=1) for group, group_df in frame.groupby('quad')]
```